Remove print() and isolate demo code fragments in reference implementation

awwad opened the pull request:
This PR addresses two related issues: `print()` and demo code in the reference implementation. In particular:
1. It strips some print() statements that lingered in two modules in the Reference Implementation and replaces them with logging commands (`logger.debug()`, `logger.info()`, `logger.warn()`, `logger.error()`). Print statements are not generally appropriate in the Reference Implementation.
2. It isolates two pieces of demo code that sit inside the Reference Implementation (in primary.py) because of limitations in the workflow when it comes to notifying viewers of the demo about exactly what's going on (When a file is rejected by the Primary, the Primary normally moves on to the next image without much fanfare). This is done by adding a variable to uptane/__init__.py, DEMO_MODE, which is set to False by default and set to True by the demo component modules (`demo_primary.py` etc). The Primary's DEFENDED banners displaying will only occur in DEMO_MODE.

More logging should be added in the future, but that is outside of the scope of this PR.

awwad committed:
In primary.py, replace print() with logging. Wedge demo code in conditional

so that it only triggers when uptane.DEMO_MODE is True. Add that variable
to uptane/__init__.py to enable this, and turn it on in each of the
demo modules.

awwad committed:
In timeserver.py, replace print() with log statement

vladimir-v-diaz commented on uptane/services/timeserver.py:
```diff
@@ -19,7 +19,7 @@
  import pyasn1.type
 except ImportError:
- print('Minor: pyasn1 library not found. Proceeding using JSON only.')
+ uptane.logger.error('pyasn1 library not found. Proceeding using JSON only.')
```
👍 
I verified that print() is not used by the other modules in the reference implementation, and only found its use in the demo code.

vladimir-v-diaz commented on demo/demo_director.py:
```diff
@@ -53,6 +53,11 @@
 import atexit # to kill server process on exit()
 
+
+# Provided for consistency; used only by demo_primary currently.
```
After reading this comment on its own, I was left wondering if setting `uptane.DEMO_MODE` to True causes special behavior in the reference implementation, or if demo code uses it as a flag to do alternative behavior of its own.

awwad replied:
Ah. True, I'll clarify.

vladimir-v-diaz commented on uptane/clients/primary.py:
```diff
@@ -619,37 +620,41 @@ def primary_update_cycle(self):
+              text='No image was found that exactly matches the signed metadata '
+              'from the Director and Image Repositories. Not keeping '
+              'untrustworthy files. ' + repr(target_filepath), sound=TADA)
+          import time
```
From our code style guidelines:

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

vladimir-v-diaz replied:
I saw `import time` multiple times on separate lines.

awwad replied:
Yes - see comment in the other PR, [here](https://github.com/uptane/uptane/pull/133#discussion_r139813407).

awwad replied:
Gonna fix both anyway - I'll have the demo-conditional import at the top and make the comments there.

vladimir-v-diaz commented on uptane/services/timeserver.py:
```diff
@@ -19,7 +19,7 @@
- print('Minor: pyasn1 library not found. Proceeding using JSON only.')
+ uptane.logger.error('pyasn1 library not found. Proceeding using JSON only.')
 else:
  #import uptane.ber_encoder as ber_encoder
```
Outdated code?

awwad replied:
Why do you ask? (BTW, I haven't really done much testing lately of the behavior when pyasn1 isn't available. It's something I have to figure out.)

vladimir-v-diaz replied:
Sorry, I mean that line 24 is commented out.  It's also an unexpected place to import a module.

awwad replied:
Ah! Yes, I think this PR is a good place to finally get rid of that, thanks..... (:

awwad committed:
DOC: Clarify explanation of DEMO_MODE setting in the demo.

awwad committed:
DOC: Remove finished TODO and outdated comment in timeserver.py

awwad committed:
Pull demo-mode-only imports to the top of primary.py

and always import them, so as to avoid the use of conditional logic
in the import section.
